### PR TITLE
Use officer_full_name in the case table

### LINKED ISSUE #1213: Use new `officer_full_name` column for Lantern dashboard
**What needs to be done? Why does it need to be done?**
PR https://github.com/Recidiviz/recidiviz-data/pull/8605 is going to remove the names from the `agent_external_id` field, which in turn will remove the names from the `officer` columns currently being sent to the Lantern dashboard.

PR https://github.com/Recidiviz/recidiviz-data/pull/8656 adds a new `officer_full_name` column to the `revocations_matrix_filtered_caseload` view. The FE needs to be updated to use this new column in the "Agent/Officer" column in the filtered caseload so that we can merge the PR that removes the name from the `officer` column. The `officer` column will then just be the unique state identifier of the agent (e.g. `E091983`).

**Additional context**
Depends on https://github.com/Recidiviz/recidiviz-data/pull/8656 being merged an in a deploy.

### PR DESCRIPTION
**Pull request by daschi**
## Description of the change

This PR is dependent on this change in the backend getting into staging so we can re-export the staging data for the case table without the commas in the officer's name (https://github.com/Recidiviz/recidiviz-data/pull/8724). 

This updates the case table in Lantern to use the column `officer_full_name` instead of parsing the `officer` column for the name. The `officer` column will still be used as a unique identifier for the row. 

Next steps:
- [x] BE recidiviz-data deployed with comma's removed from officer_full_name (https://github.com/Recidiviz/recidiviz-data/pull/8724)
- [ ] This PR merged and deployed to staging
- [ ] PR #8656 is deployed in production (can be a cherry pick) (https://github.com/Recidiviz/recidiviz-data/pull/8656)
- [ ] FE deployed with officer_full_name handling (in production)
- [ ] Merge Emily’s [#8605 PR ](https://github.com/Recidiviz/recidiviz-data/pull/8605)

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

> Closes #1213

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing against realistic data has been performed locally
- [ ] E2E have been run for Lantern changes ([Steps in the Readme](https://github.com/Recidiviz/pulse-dashboard#running-e2e-tests))

### Code review

These boxes should be checked by reviewers prior to merging:

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [x] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: src/lantern/CaseTable/__tests__/useSort.test.js [hunk 1]
```diff
@@ -23,7 +23,7 @@ describe("useSort", () => {
     {
       state_id: "123",
       district: "01",
-      officer: "333: Callie Torres",
+      officer: "Callie Torres",
```
Comment (jovergaag) [inline src/lantern/CaseTable/__tests__/useSort.test.js:26]: OMG. I'm just seeing these names for the first time. LOL. I love you.

**Review by jovergaag — APPROVED** [review]
This looks good.

It looks like the officer names in `revocations_matrix_distribution_by_officer` did not change. Was this change intended to ripple out to all officer names do you know? Officer chart looks fine, I'm just curious.

**Comment by daschi** [comment]
> It looks like the officer names in revocations_matrix_distribution_by_officer did not change.

Yeah that's right! We just needed to update the column being used to create the `officer_label` column for `revocations_matrix_distribution_by_officer`, so no FE change needed :)

File: src/lantern/CaseTable/__tests__/useSort.test.js (hunk 1 above)
Comment (Catacola) [reply src/lantern/CaseTable/__tests__/useSort.test.js:26]: I haven't seen that name in forever. Now I feel like I need to see a bone set.

**Review by jjdressel11 — APPROVED** [review]

**Pull request merged**